src/role_claimer.js

```diff
@@ -22,7 +22,7 @@ roles.claimer.action = function(creep) {
   const returnCode = creep.claimController(creep.room.controller);
   if (returnCode === OK) {
     creep.creepLog('New claimer, in room, claimed');
-    creep.suicide();
+    Creep.recycleCreep(creep);
```

Comment:
this should depend on 
```js 
if (creep.memory.timeToTravel > 0 && creep.memory.born - Game.time < creep.memory.timeToTravel) {
// creep will not reach spawn in time
  creep.suicide();
} else {
  Creep.recycleCreep(creep);
}
```

Reply:
I will refine the existed recycleCreep function, and the check would be added there, using getRoomLinearDistance.